### PR TITLE
docs(markdown-navigator): add several demos

### PR DESCRIPTION
**Pull request by richavyas**
## Description

Bring editor to the demo for markdown-navigator from old docs since it's quite a useful feature to do in-place .md file testing for IPH

takes care of #1636 

### What's included?
- editor in examples section
- shows two examples with two different cards
- basic demo is used as hero
- styling clean up

#### Test Steps
- [ ] `npm run serve`
- [ ] navigate to http://localhost:3000/#/components/markdown-navigator
- [ ] click on examples and test if . editor is working as expected

#### General Tests for Every PR

- [ ] `npm run serve:prod` still works.
- [ ] `npm run tslint` passes.
- [ ] `npm run stylelint` passes.
- [ ] `npm test` passes and code coverage is not lower.
- [ ] `npm run build:lib` still works.

##### Screenshots or link to StackBlitz/Plunker

NavDemo

### CODE REVIEW

File: src/app/content/components/component-demos/markdown-navigator/demos/markdown-navigator-demo-editor/markdown-navigator-demo-editor.component.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  },
+  biLevelTree[1],
+  biLevelTree[2],
+];
+
+@Component({
+  selector: 'markdown-navigator-demo-editor',
+  styleUrls: ['./markdown-navigator-demo-editor.component.scss'],
+  templateUrl: './markdown-navigator-demo-editor.component.html',
+  preserveWhitespaces: true,
```
Comment (christianmemije) [inline src/app/content/components/component-demos/markdown-navigator/demos/markdown-navigator-demo-editor/markdown-navigator-demo-editor.component.ts]: No need

**Comment by christianmemije** [comment]
Overflowing
Screen Shot 2020-03-06 at 12 33 41 PM

**Comment by christianmemije** [comment]
Need to fix responsiveness 
Screen Shot 2020-03-06 at 12 34 56

File: src/app/content/components/component-demos/markdown-navigator/demos/markdown-navigator-demo-editor/markdown-navigator-demo-editor.component.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
+
+<div class="push-bottom" layout-xs='column' layout-sm="column" layout-gt-sm='row' layout-align='center center' [style.height.px]="450">
+  <div layout='column' flex="40" flex-xs='100' flex-order='2'flex-order-gt-sm='1' class='pad-bottom-lg'>
+    <div class="pad-top push-bottom">
+      <button mat-raised-button class="text-upper" color="primary" (click)="applyInput()">Try</button>
+      <span class="mat-hint">Try the markdown and open in a dialog</span>
+    </div>
+    <div>
+      <button mat-raised-button class="text-upper" color="primary" (click)="demoStartAt()">Jump</button>
+      <span class="mat-hint">Jump to an item in the markdown</span>
```
Comment (christianmemije) [inline src/app/content/components/component-demos/markdown-navigator/demos/markdown-navigator-demo-editor/markdown-navigator-demo-editor.component.html]: Jump to an item within the tree*

File: src/app/content/components/component-demos/markdown-navigator/demos/markdown-navigator-demo-editor/markdown-navigator-demo-editor.component.html [hunk 3]
```diff
@@ -0,0 +1,22 @@
+
+<div class="push-bottom" layout-xs='column' layout-sm="column" layout-gt-sm='row' layout-align='center center' [style.height.px]="450">
+  <div layout='column' flex="40" flex-xs='100' flex-order='2'flex-order-gt-sm='1' class='pad-bottom-lg'>
+    <div class="pad-top push-bottom">
+      <button mat-raised-button class="text-upper" color="primary" (click)="applyInput()">Try</button>
+      <span class="mat-hint">Try the markdown and open in a dialog</span>
```
Comment (christianmemije) [inline src/app/content/components/component-demos/markdown-navigator/demos/markdown-navigator-demo-editor/markdown-navigator-demo-editor.component.html]: Both buttons open in a dialog so redundant.

File: src/app/content/components/component-demos/markdown-navigator/markdown-navigator.model.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import { IMarkdownNavigatorItem } from '@covalent/markdown-navigator';
```
Comment (christianmemije) [inline src/app/content/components/component-demos/markdown-navigator/markdown-navigator.model.ts]: model files tend to have interfaces not data. Also only export what is necessary.

File: src/app/content/components/component-demos/markdown-navigator/demos/markdown-navigator-demo-editor/markdown-navigator-demo-editor.component.ts (hunk 1 above)
Comment (thomaslennan) [reply src/app/content/components/component-demos/markdown-navigator/demos/markdown-navigator-demo-editor/markdown-navigator-demo-editor.component.ts]: Fixed.

**Comment by christianmemije** [comment]
I'll be taking over this PR

**Comment by christianmemije** [comment]
Superseded by #1738

**Pull request closed without merging**